src/hooks/spaces/useWorldSpaces.ts

```diff
@@ -0,0 +1,29 @@
+import { useFirestore, useFirestoreCollectionData } from "reactfire";
+import { collection, query, where } from "firebase/firestore";
+
+import { ALWAYS_EMPTY_ARRAY, COLLECTION_SPACES } from "settings";
+
+import { SpaceId, SpaceWithId, SpaceWithoutId, WorldId } from "types/id";
+
+import { withIdConverter } from "utils/converters";
+
+type UseWorldSpaces = (options: {
+  worldId: WorldId;
+}) => { spaces: SpaceWithId[] };
+
+export const useWorldSpaces: UseWorldSpaces = ({ worldId = "" }) => {
```

Comment:
With all the new HOCs we have we should not need to have a default here. Calling this without a worldId specified is a coding error and should fail early.

Reply:
Done in https://github.com/sparkletown/sparkle/pull/2820/commits/af825f995a7267c0da1cdbc23ea8630055576519